Plugin/CheckoutLayoutPlugin.php

```diff
@@ -0,0 +1,76 @@
+
+
+class CheckoutLayoutPlugin
+{
+    public function __construct(
+        ScopeSetting $klaviyoScopeSetting
+    ) {
+        $this->_klaviyoScopeSetting = $klaviyoScopeSetting;
+    }
+
+    public function afterProcess(\Magento\Checkout\Block\Checkout\LayoutProcessor $processor, $jsLayout)
+    {
+        if ($this->_klaviyoScopeSetting->getConsentAtCheckoutSMSIsActive())
+        {
+            $smsConsentCheckbox = [
+                'component' => 'Magento_Ui/js/form/element/abstract',
+                'config' => [
+                    'customScope' => 'shippingAddress.custom_attributes',
+                    'template' => 'ui/form/field',
+                    'elementTmpl' => 'ui/form/element/checkbox',
+                    'options' => [],
+                    'id' => 'kl_sms_consent',
+                ],
+                'dataScope' => 'shippingAddress.custom_attributes.kl_sms_consent',
+                'label' => 'Sign Up for SMS Consent',
+                'description' => $this->_klaviyoScopeSetting->getConsentAtCheckoutSMSConsentText(),
+                'provider' => 'checkoutProvider',
+                'visible' => true,
+                'checked' => false,
+                'validation' => [],
+                'sortOrder' => $this->_klaviyoScopeSetting->getConsentAtCheckoutSMSConsentSortOrder(),
+                'id' => 'kl_sms_consent',
+            ];
+
+            $jsLayout['components']['checkout']['children']['steps']['children']['shipping-step']['children']['shippingAddress']['children']['shipping-address-fieldset']['children']['kl_sms_consent'] = $smsConsentCheckbox;
+        }
+        // Open to ideas here, since we don't overwrite the customer-email section
+        // we need to distinguish if the customer is logged in or not, object manager is an easy way to do so
+        $objectManager = \Magento\Framework\App\ObjectManager::getInstance();
+        $customerSession = $objectManager->get('Magento\Customer\Model\Session');
+
+        if (!$customerSession->isLoggedIn() && $this->_klaviyoScopeSetting->getConsentAtCheckoutEmailIsActive())
+        {
+            $emailConsentCheckbox = [
+                'component' => 'Magento_Ui/js/form/element/abstract',
+                'config' => [
+                    'customScope' => 'shippingAddress.custom_attributes',
+                    'template' => 'ui/form/field',
+                    'elementTmpl' => 'ui/form/element/checkbox',
+                    'options' => [],
+                    'id' => 'kl_email_consent',
+                ],
+                'dataScope' => 'shippingAddress.custom_attributes.kl_email_consent',
+                'label' => 'Sign Up for Email Newsletters',
+                'description' => $this->_klaviyoScopeSetting->getConsentAtCheckoutEmailConsentText(),
+                'provider' => 'checkoutProvider',
+                'visible' => true,
+                'checked' => false,
+                'validation' => [],
+                'sortOrder' => 0,
```

Comment:
im curious how the email consent doesn't need a sortOrder setting and yet appears in the right place for a checkout yet sms consent takes default value of 200 and is configurable and seems to work

Reply:
great question, i thought about adding it, it's basically because the email form will always be at the top of the page, this is inside the shipping address form which will be rendered after the email form, so putting it at top makes sense.  For the phone number, 200 is just arbitrary to make sure it's at the bottom, but customers CAN customize the sort order of the shipping fields and if they did, phone number could be moved so we should allow them to move the consent box as well